### DISCUSSION TITLE
Warp and Kaolin ?

### DISCUSSION THREAD

**Discussion by ucacaxm** [NVIDIA/warp#352] [General]
Hi,

I just discover Warp, it seems cool. I used a bit Taishi, so I understand the idea behind Warp and like the concept of kernel.
Just I am a bit confuse with Kaolin. Kaolin include an "Agnostic Physics Simulation" and "Physics Simulation for Points, Meshes, Splats and more".

For instance, let take two examples:
1) I want to simulate a deformable creature and plug it with an RL code (for instance PPO) to learn how to move.
There are a similar example with the quadruped, but the same thing with a deformable creature.

2) if I want to perform a skinning (LBS) on a 3D mesh linked to a skeleton and optimize some part of the skeleton or the mesh. 
There is no particular physics in this example but the autodiff aspect is important, and I need to load a mesh and optimize the vertices.

Do I better use Warp of Kaolin ?

Many thanks,

**Comment by mmacklin** [comment]
I think you could likely use Warp in both cases.

For 1., you may want to look at our deformable bear / walker example: https://github.com/NVIDIA/warp/blob/main/warp/examples/optim/example_walker.py

For 2. I'm not sure if Kaolin provides this, but you could quite easily write an LBS kernel in Warp and run it through auto-diff, however I don't have an example of this out-of-the-box.

Cheers,
Miles
